### ISSUE TITLE
Define compact inspection proof evidence contract for agent-facing verdicts

### ISSUE THREAD

**Issue by cbusillo** [cbusillo/jetbrains-inspection-api#114]
## Summary

Define the compact machine-readable proof contract that lets the plugin/API, `jb-inspect.py`, and Code exec-harness agree on when a JetBrains inspection result is proven `GREEN`, proven `RED`, or must be `UNKNOWN`.

This is part of the false-green workstream tracked by #113 and paired with helper hardening in cbusillo/codex-skills#388.

## Why

The dangerous failure mode is not that an IDE inspection can fail. The dangerous failure mode is converting an unproven run into `GREEN` because `total_problems == 0`.

We need the scripts to see enough compact proof to reject false green states without sending verbose IDE diagnostics into normal LLM context.

## Current Status

New evidence from the Code exec harness proves the desired agent-facing behavior: contradictory `GREEN`/`0` evidence must be reported as `UNKNOWN`, not clean. A helper branch in `codex-skills` also proves/fixes nested `.idea` project path collapse and `proof_failures` overriding plugin-provided `GREEN`.

This issue owns the product/API contract for that proof evidence.

## Proposed Inline Contract

Inline helper/API responses should expose a compact bounded proof summary, such as:

```json
{
  "verdict": "GREEN|RED|UNKNOWN",
  "verdict_reason": "clean_confirmed|actionable_findings|inspection_proof_failed|...",
  "proof_failures": [],
  "proof": {
    "requested_path": "...",
    "resolved_project_path": "...",
    "route_base_path": "...",
    "profile": "...",
    "expected_profile": "...",
    "scope": "changed_files",
    "scope_file_count": 7,
    "session_fresh": true,
    "indexing_complete": true,
    "capture_complete": true,
    "inspection_completed": true,
    "inspection_ids_registered": ["OdooPyInspection", "OdooXmlInspection"]
  },
  "diagnostic_artifact": "path-or-null"
}
```

The exact schema can change during implementation, but the contract should stay compact and script-first.

## Artifact-Only Diagnostics

Do not put large diagnostics in ordinary agent context:

- full `capture_diagnostic` trees
- raw inspection tree/model dumps
- full plugin lists
- full route inventories
- full transcripts
- unbounded timestamps/logs/problems

Those belong in local artifacts/logs referenced by `diagnostic_artifact` when deeper debugging is needed.

## Acceptance Criteria

- API/helper responses can prove route identity, selected project/worktree, session freshness, requested scope/filter, lifecycle completion, capture completeness, profile identity, and expected/registered inspection IDs with compact bounded fields.
- Contradiction codes cover states like `route_mismatch`, `profile_mismatch`, `empty_scope`, `missing_inspection_ids`, `stale_session`, `capture_incomplete`, `timeout`, `indexing`, and `cleanup_failed`.
- `proof_failures` makes the result non-clean/UNKNOWN regardless of `total_problems: 0`.
- Ordinary agent-facing output includes only verdict, reason, compact proof/failure summary, truncated actionable findings, and optional artifact reference.
- Verbose diagnostics stay artifact-only unless a debugging flow explicitly asks for them.
- The contract is consumed by cbusillo/codex-skills#388 and covered by the Code exec-harness false-green fixture in cbusillo/code.

Refs #113
Refs cbusillo/codex-skills#388

**Comment by cbusillo** [comment]
Current status update:

PR #115 implements the compact proof contract from this issue and is now green/mergeable.

What landed in the PR:

- API responses include compact `inspection_proof` and `proof_failures` fields.
- `proof_failures` can block a `GREEN` verdict and force `UNKNOWN`.
- `/api/inspection/problems` includes current run state before proof/verdict calculation, so an old clean snapshot cannot look green while a newer inspection is active.
- MCP output puts verdict guidance before JSON and removes misleading zero-result fields from ordinary `UNKNOWN` payloads, while preserving explicit `include_stale=true` cached diagnostics.

Validation on the PR branch passed locally:

```text
./gradlew test :mcp-server-jvm:test --tests com.shiny.inspectionmcp.InspectionSnapshotStateTest --tests com.shiny.inspectionmcp.InspectionHandlerTest --tests com.shiny.inspectionmcp.mcpserver.McpServerTest
./gradlew :test
./gradlew :inspection-core:test
./gradlew :mcp-server-jvm:test
./gradlew buildPlugin
```

GitHub checks are also green: 4 passed, 0 failed, 0 pending.

Next after merge: use #113 to verify the WebStorm red-lane behavior against a live IDE/plugin build and confirm the system reports `RED` or `UNKNOWN`, never false `GREEN`.

**Comment by cbusillo** [comment]
Final status: the compact proof contract from this issue is complete and has been exercised by the full WebStorm red-lane workstream.

What landed:

- PR #115 added compact `inspection_proof` / `proof_failures` fields and prevented contradictory proof from becoming `GREEN`.
- PR #116 then fixed the remaining WebStorm red-lane empty-model path while preserving that proof contract.
- The parent #113 has now been closed after post-merge WebStorm validation from `main`.

Final live proof from merged `main`:

```text
plugin_build_fingerprint=91ace601dbf7-clean
status=ok
bucket=red_confirmed
verdict=RED
total=2
cleanup=closed
```

This confirms the contract does its intended job: ambiguous/contradictory inspection evidence stays non-clean, and the fixed WebStorm proof path can now produce actionable RED evidence.

**Comment by cbusillo** [comment]
Closing as complete: PR #115 implemented the compact proof contract, PR #116 preserved it while fixing WebStorm red-lane, and #113 closed after merged-main RED proof.
